os/arch/arm/src/armv8-m/up_assertlogs.c

```diff
@@ -0,0 +1,570 @@
+
+#ifdef CONFIG_STACK_COLORATION
+static inline void up_showtasks(void)
+{
+	lldbg("*******************************************\n");
+	lldbg("List of all tasks in the system:\n");
+	lldbg("*******************************************\n");
+
+#if CONFIG_TASK_NAME_SIZE > 0
+        lldbg("%*s | %5s | %4s | %5s | %7s / %7s | %16p | %20s | %8s\n", CONFIG_TASK_NAME_SIZE, "NAME", "PID", "PRI", "TYPE", "USED", "TOTAL STACK", "STACK ALLOC ADDR", "STACK RANGE", "CURRENT SP");
+        lldbg("--------------------------------------------------------------------------------------------------------------------\n");
+#else
+	lldbg("%5s | %4s | %7s / %7s | %16s\n", "PID", "PRI", "USED", "TOTAL STACK", "STACK ALLOC ADDR");
+	lldbg("------------------------------------------------------\n");
+#endif
+
+	/* Dump interesting properties of each task in the crash environment */
+
+	sched_foreach(up_taskdump, NULL);
+}
+#else
+#define up_showtasks()
+#endif
+
+
+/****************************************************************************
+ * Name: assert_tracecallback
+ ****************************************************************************/
+
+#ifdef CONFIG_ARCH_USBDUMP
+static int usbtrace_syslog(FAR const char *fmt, ...)
+{
+	va_list ap;
+	int ret;
+
+	/* Let vsyslog do the real work */
+
+	va_start(ap, fmt);
+	ret = lowvsyslog(LOG_INFO, fmt, ap);
+	va_end(ap);
+	return ret;
+}
+
+static int assert_tracecallback(FAR struct usbtrace_s *trace, FAR void *arg)
+{
+	usbtrace_trprintf(usbtrace_syslog, trace->event, trace->value);
+	return 0;
+}
+#endif
+
+/****************************************************************************
+ * Name: up_dumpstate
+ ****************************************************************************/
+
+#ifdef CONFIG_ARCH_STACKDUMP
+static void up_dumpstate(void)
+{
+	struct tcb_s *rtcb = this_task();
+	uint32_t sp = up_getsp();
+	uint32_t ustackbase;
+	uint32_t ustacksize;
+#if CONFIG_ARCH_INTERRUPTSTACK > 3
+	uint32_t istackbase;
+	uint32_t istacksize;
+#endif
+
+	/* Get the limits on the user stack memory */
+
+	ustackbase = (uint32_t)rtcb->adj_stack_ptr;
+	ustacksize = (uint32_t)rtcb->adj_stack_size;
+
+#if CONFIG_ARCH_INTERRUPTSTACK > 3
+	/* Get the limits on the interrupt stack memory */
+
+	istackbase = (uint32_t)&g_intstackbase;
+	istacksize = (CONFIG_ARCH_INTERRUPTSTACK & ~3);
+
+	/* Show interrupt stack info */
+
+	lldbg("sp:     %08x\n", sp);
+	lldbg("IRQ stack:\n");
+	lldbg("  base: %08x\n", istackbase);
+	lldbg("  size: %08x\n", istacksize);
+#ifdef CONFIG_STACK_COLORATION
+	lldbg("  used: %08x\n", up_check_intstack());
+#endif
+
+	/* Does the current stack pointer lie within the interrupt
+	 * stack?
+	 */
+
+	if (sp <= istackbase && sp > istackbase - istacksize) {
+		/* Yes.. dump the interrupt stack */
+
+		up_stackdump(sp, istackbase);
+	}
+
+	/* Extract the user stack pointer if we are in an interrupt handler.
+	 * If we are not in an interrupt handler.  Then sp is the user stack
```

Comment:
2nd comment line
(If..., then...)
Could you modify for comment sentence continuity please?

Reply:
This file was added by mistake. Have removed it now